tempel.el

```diff
@@ -348,10 +363,21 @@ PROMPT is the optional prompt/default value."
 
 (defun tempel--save ()
   "Save template file buffer."
-  (when-let (buf (get-file-buffer tempel-file))
-    (with-current-buffer buf
-      (when (and (buffer-modified-p) (y-or-n-p (format "Save file %s? " tempel-file)))
-        (save-buffer buf)))))
+  (let (modified)
+    (dolist (file (tempel--expand-locations tempel-paths))
+      (when-let (buff (get-file-buffer tempel-file))
+  	(push file modified)))
+    (when (or (and (= (length modified) 1)
+		   (y-or-n-p "Save template file %s? " (car modified)))
+	      (y-or-n-p "Save modified template files? ")))
+    (mapc #'save-buffer modified)))
+
+(defun tempel--paths-modified-p ()
+  "Return non-nil if any template files have been modified since loading."
+  (thread-last (tempel--expand-locations tempel-paths)
+    (mapcar (lambda (file) (file-attribute-modification-time (file-attributes file))))
+    (apply #'max)
+    (< tempel--paths-modified)))
```

Comment:
This is almost good but will fail if there are zero paths. `max` needs at least one argument. So you should write `(apply #'max 0)`.